opm/simulators/flow/FIBlackoilModel.hpp

```diff
@@ -59,23 +63,47 @@ class FIBlackOilModel : public BlackOilModel<TypeTag>
     FIBlackOilModel(Simulator& simulator)
         : BlackOilModel<TypeTag>(simulator)
     {
+        const auto& gv = this->gridView_;
+#ifdef _OPENMP
+        const int nt = omp_get_max_threads();
+        if (nt > 1) {
+            const auto num_elements = gv.size(0);
+            constexpr int max_chunk_size = 1000;
+            const int chunk_size = std::max(1, std::min(max_chunk_size, num_elements / nt));
```

Comment:
Oh it's a ~Unix system~ [`clamp()`](https://en.cppreference.com/w/cpp/algorithm/clamp).  I know this:
```C++
const auto chunk_size = std::clamp(num_elements / nt, 1, max_chunk_size);
```